recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,103 @@
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "apple-clang": "9.4",
+            "clang": "3.3",
+            "gcc": "4.9.4"
+        }
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        compiler = self.settings.compiler
+        min_version = self._minimum_compilers_version[str(compiler)]
+        try:
+            if tools.Version(compiler.version) < min_version:
+                msg = (
+                    "{} requires C{} features which are not supported by compiler {} {} !!"
+                ).format(self.name, self._minimum_cpp_standard, compiler, compiler.version)
+                raise ConanInvalidConfiguration(msg)
+        except KeyError:
+            msg = (
+                "{} recipe lacks information about the {} compiler, "
+                "support for the required C++{} features is assumed"
+            ).format(self.name, compiler, self._minimum_cpp_standard)
+            self.output.warn(msg)
+
+        if (not (self.version in self.conan_data["sources"]) or
+            not (str(self.settings.os) in self.conan_data["sources"][self.version]) or
+            not (str(self.settings.arch) in self.conan_data["sources"][self.version][str(self.settings.os)] ) ):
+            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version][str(self.settings.os)][str(self.settings.arch)], destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        include_path = os.path.join(self._source_subfolder, 'include')
+        self.copy('*.h', dst='include', src=include_path)
+        self.copy('*.hh', dst='include', src=include_path)
+        self.copy('*.hpp', dst='include', src=include_path)
+
+        self.copy('*.lib', dst='lib', keep_path=False)
+        self.copy('*.dll', dst='bin', keep_path=False)
+        self.copy('*.so', dst='lib', keep_path=False)
+        self.copy('*.dylib', dst='lib', keep_path=False)
+        self.copy('*.a', dst='lib', keep_path=False)
+
+        self.copy('LICENSE', dst='licenses', src=self._source_subfolder)
+
+    def package_info(self):
+        if self.options.shared:
+            if self.settings.os == "Windows":
+                self.cpp_info.libs = ["wasmtime.dll"]
+            else:
+                self.cpp_info.libs = ["wasmtime"]
+        else:
+            if self.settings.os == "Windows":
+                self.cpp_info.defines= ["/DWASM_API_EXTERN=", "/DWASI_API_EXTERN="]
+            self.cpp_info.libs = ["wasmtime"]
```

Comment:
When you do the rename of the import library in https://github.com/conan-io/conan-center-index/pull/6083/files#r670899230, then these lines can become:
```suggestion
        self.cpp_info.libs = ["wasmtime"]
        if self.settings.os == "Windows" and self.options.shared:
            self.cpp_info.defines.extend(["WASM_API_EXTERN", "WASI_API_EXTERN"])
```

Reply:
You miss `wasmtime.dll` in your suggestion

This package has library like this `wasmtime.dll.lib`

Reply:
I've updated my suggestion.

Reply:
`"WASM_API_EXTERN", "WASI_API_EXTERN"` soes not work on Windows

Reply:
You're right. The `_EXTERN` are used directly in front of the function.
`WASI_API_EXTERN` is used here: https://github.com/bytecodealliance/wasmtime/blob/6f3adacb9fb37043d77279b90e63517b07628011/crates/c-api/include/wasi.h#L12-L18

I've updated the suggestion again.

Reply:
I do not see any updates ...
Only way it works like this:
"/DWASM_API_EXTERN="
"WASI_API_EXTERN"

Reply:
Consider https://github.com/redradist/conan-center-index/pull/1/files#diff-3ae4adc5301f1c79ce32b6b1dc1073b31229e7ff3c050955bb414a290c58707fR82 I tested and worked.

Reply:
It is not translated to proper MSVC flag

Reply:
What client version and generator are you using?

I use this  and it works perfectly
https://github.com/conan-io/conan-center-index/blob/467c8331e67f6aa1798a0a3c4a0a008e27e42c6a/recipes/cpprestsdk/all/conanfile.py#L125-L128